File: mobile/android/app/src/main/kotlin/ai/pazl/alma/ui/screens/PeopleScreen.kt
package ai.pazl.alma.ui.screens

import ai.pazl.alma.R
import ai.pazl.alma.core.AppContainer
import ai.pazl.alma.data.ApiResult
import ai.pazl.alma.data.dto.BirthInput
import ai.pazl.alma.data.dto.PlaceDto
import ai.pazl.alma.data.dto.ProfileDto
import ai.pazl.alma.ui.components.GoldButton
import ai.pazl.alma.ui.components.QuietButton
import ai.pazl.alma.ui.sky.NightSky
import ai.pazl.alma.ui.sky.SkyConfig
import ai.pazl.alma.ui.theme.AlmaPalette
import ai.pazl.alma.ui.theme.AlmaTheme
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.input.ImeAction
import androidx.compose.ui.text.input.KeyboardCapitalization
import androidx.compose.ui.unit.dp
import androidx.lifecycle.ViewModel
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import androidx.lifecycle.viewModelScope
import androidx.lifecycle.viewmodel.compose.viewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.launch

/**
 * The people this account has saved. Compatibility needs at least one.
 *
 * **This was the last stub in the app**, and what it said was worse than
 * nothing: "adding a second person is not built on Android yet — add one on the
 * web and it will be here". A screen that sends somebody to a different product
 * to finish a job is a screen that has given up, and compatibility is one of
 * the eight systems this app sells.
 *
 * iOS has had the real thing for a while and this is the same screen from the
 * same three calls — list, save, delete. What it deliberately does not do is
 * compute anything: every row is what was typed, so there is nothing here that
 * can be wrong.
 */
@Composable
fun PeopleScreen(
    container: AppContainer,
    onBack: () -> Unit,
) {
    val vm: PeopleViewModel = viewModel { PeopleViewModel(container) }
    val state by vm.state.collectAsStateWithLifecycle()

    LaunchedEffect(Unit) { vm.load() }

    NightSky(config = SkyConfig(seed = 9)) {
        CabinetPage {
            ScreenTitle(stringResource(R.string.people_title), onBack = onBack)

            if (state.adding) {
                AddPerson(
                    vm = vm,
                    onCancel = vm::stopAdding,
                )
                return@CabinetPage
            }

            Text(
                text = stringResource(R.string.people_lead),
                style = AlmaTheme.type.dayVoice,
                modifier = Modifier.padding(top = 10.dp),
            )

            Spacer(Modifier.height(24.dp))

            if (state.people.isEmpty()) {
                Text(
                    text = stringResource(R.string.people_none_yet),
                    style = AlmaTheme.type.meta,
                )
                Spacer(Modifier.height(18.dp))
            } else {
                RuledLabel(
                    stringResource(R.string.people_saved),
                    trailing = state.people.size.toString(),
                )
                state.people.forEachIndexed { index, person ->
                    PersonRow(
                        person = person,
                        last = index == state.people.lastIndex,
                        onRemove = { vm.remove(person) },
                    )
                }
                Spacer(Modifier.height(22.dp))
            }

            GoldButton(
                text = stringResource(R.string.cabinet_add_person),
                onClick = vm::startAdding,
                modifier = Modifier.fillMaxWidth(),
            )

            // Whose birth data this is, and what we did about asking. The terms
            // ask the person entering it to have asked first, and saying so at
            // the point of entry is worth more than saying it in a document.
            Text(
                text = stringResource(R.string.people_consent),
                style = AlmaTheme.type.meta,
                color = AlmaPalette.Muted3,
                modifier = Modifier.padding(top = 24.dp),
            )

            state.error?.let {
                Text(
                    text = it,
                    style = AlmaTheme.type.meta,
                    color = AlmaPalette.Disagree,
                    modifier = Modifier.padding(top = 16.dp),
                )
            }
        }
    }
}

/**
 * One saved birth. Everything shown is what was entered.
 *
 * Removing asks twice, and that is not ceremony: every reading keyed to this
 * person goes with them, and those are chapters somebody paid for.
 */
@Composable
private fun PersonRow(person: ProfileDto, last: Boolean, onRemove: () -> Unit) {
    var confirming by remember { mutableStateOf(false) }
    CabinetRow(rule = !last) {
        Column(Modifier.weight(1f)) {
            Text(
                text = person.name?.takeIf { it.isNotBlank() }
                    ?: stringResource(R.string.people_unnamed),
                style = AlmaTheme.type.headingM,
            )
            Text(
                text = listOfNotNull(
                    person.birthDate,
                    person.birthTime ?: stringResource(R.string.people_time_unknown),
                    person.placeLabel,
                ).joinToString(" · "),
                style = AlmaTheme.type.meta,
                modifier = Modifier.padding(top = 3.dp),
            )
        }
        QuietButton(
            text = if (confirming) {
                stringResource(R.string.people_remove_confirm)
            } else {
                stringResource(R.string.people_remove)
            },
            onClick = { if (confirming) onRemove() else confirming = true },
            contentColor = AlmaPalette.Disagree,
        )
    }
}

/**
 * A second birth, asked for in the journey's own order and with its own fields.
 *
 * Shorter than the journey by one thing and it is the ceremony: nothing is
 * being revealed here, a comparison is being enabled.
 *
 * **`is_self = false` is explicit and must stay explicit.** "Not said" on a
 * second birth would overwrite the account owner's own chart and every reading
 * keyed to it. iOS carries the same warning at the same line.
 */
@Composable
private fun AddPerson(vm: PeopleViewModel, onCancel: () -> Unit) {
    val draft by vm.draft.collectAsStateWithLifecycle()

    Spacer(Modifier.height(18.dp))

    AlmaTextField(
        value = draft.name,
        onValueChange = vm::onName,
        placeholder = "",
        label = stringResource(R.string.journey_name_aria),
        imeAction = ImeAction.Next,
        capitalisation = KeyboardCapitalization.Words,
    )
    Spacer(Modifier.height(12.dp))

    // The date as text in the engine's own format. A wheel picker would be the
    // journey's answer, and the journey's wheels are bound to its own view
    // model; borrowing them here would mean two screens sharing one draft.
    AlmaTextField(
        value = draft.birthDate,
        onValueChange = vm::onDate,
        placeholder = "1994-03-12",
        label = stringResource(R.string.people_birth_date),
        imeAction = ImeAction.Next,
    )
    Spacer(Modifier.height(12.dp))

    AlmaTextField(
        value = draft.birthTime,
        onValueChange = vm::onTime,
        placeholder = "14:20",
        label = stringResource(R.string.people_birth_time),
        imeAction = ImeAction.Next,
    )
    Spacer(Modifier.height(12.dp))

    AlmaTextField(
        value = draft.placeQuery,
        onValueChange = vm::onPlaceQuery,
        placeholder = "",
        label = stringResource(R.string.capture_search_place),
        imeAction = ImeAction.Search,
        capitalisation = KeyboardCapitalization.Words,
    )

    // The place has to come out of the gazetteer rather than out of a text box:
    // the coordinate sets the horizon and the zone sets the instant, and text
    // that merely looks like a city is neither.
    draft.places.take(4).forEach { place ->
        Text(
            text = place.label,
            style = AlmaTheme.type.meta,
            color = if (draft.chosen?.id == place.id) AlmaPalette.GoldBright else AlmaPalette.Body,
            modifier = Modifier
                .fillMaxWidth()
                .clickable { vm.choose(place) }
                .padding(vertical = 10.dp),
        )
    }

    Spacer(Modifier.height(20.dp))
    GoldButton(
        text = stringResource(R.string.people_save),
        onClick = vm::save,
        enabled = draft.chosen != null && draft.birthDate.isNotBlank(),
        modifier = Modifier.fillMaxWidth(),
    )
    Spacer(Modifier.height(8.dp))
    QuietButton(
        text = stringResource(R.string.nav_back),
        onClick = onCancel,
        modifier = Modifier.fillMaxWidth(),
    )
}

@androidx.compose.runtime.Immutable
data class PeopleState(
    val people: List<ProfileDto> = emptyList(),
    val adding: Boolean = false,
    val error: String? = null,
)

@androidx.compose.runtime.Immutable
data class PersonDraft(
    val name: String = "",
    val birthDate: String = "",
    val birthTime: String = "",
    val placeQuery: String = "",
    val places: List<PlaceDto> = emptyList(),
    val chosen: PlaceDto? = null,
)

class PeopleViewModel(private val container: AppContainer) : ViewModel() {

    private val _state = MutableStateFlow(PeopleState())
    val state: StateFlow<PeopleState> = _state.asStateFlow()

    private val _draft = MutableStateFlow(PersonDraft())
    val draft: StateFlow<PersonDraft> = _draft.asStateFlow()

    fun load() {
        viewModelScope.launch {
            when (val result = container.client.profiles()) {
                // Only the others. The account owner's own birth is edited in
                // Settings and must never appear in a list with a "remove"
                // button beside it.
                is ApiResult.Ok -> _state.value = _state.value.copy(
                    people = result.data.filterNot { it.isSelf },
                    error = null,
                )
                is ApiResult.Err -> _state.value =
                    _state.value.copy(error = result.failure.message)
            }
        }
    }

    fun remove(person: ProfileDto) {
        viewModelScope.launch {
            when (val result = container.client.deleteProfile(person.id)) {
                is ApiResult.Ok -> {
                    load()
                    container.session.refreshProfile()
                }
                is ApiResult.Err -> _state.value =
                    _state.value.copy(error = result.failure.message)
            }
        }
    }

    fun startAdding() {
        _draft.value = PersonDraft()
        _state.value = _state.value.copy(adding = true, error = null)
    }

    fun stopAdding() { _state.value = _state.value.copy(adding = false) }

    fun onName(value: String) { _draft.value = _draft.value.copy(name = value) }
    fun onDate(value: String) { _draft.value = _draft.value.copy(birthDate = value) }
    fun onTime(value: String) { _draft.value = _draft.value.copy(birthTime = value) }

    fun onPlaceQuery(value: String) {
        _draft.value = _draft.value.copy(placeQuery = value, chosen = null)
        if (value.length < 2) {
            _draft.value = _draft.value.copy(places = emptyList())
            return
        }
        viewModelScope.launch {
            when (val result = container.client.searchPlaces(value)) {
                is ApiResult.Ok -> _draft.value = _draft.value.copy(places = result.data)
                is ApiResult.Err -> Unit  // a failed search is an empty list, not an error banner
            }
        }
    }

    fun choose(place: PlaceDto) {
        _draft.value = _draft.value.copy(chosen = place, placeQuery = place.label, places = emptyList())
    }

    fun save() {
        val draft = _draft.value
        val place = draft.chosen ?: return
        viewModelScope.launch {
            val input = BirthInput(
                birthDate = draft.birthDate.trim(),
                birthTime = draft.birthTime.trim().ifBlank { null },
                latitude = place.latitude,
                longitude = place.longitude,
                timezone = place.timezone,
                placeLabel = place.label,
                placeId = place.id,
                name = draft.name.trim().ifBlank { null },
                // **Explicit, always.** See the note on `AddPerson`.
                isSelf = false,
            )
            when (val result = container.client.saveProfile(input)) {
                is ApiResult.Ok -> {
                    _state.value = _state.value.copy(adding = false, error = null)
                    load()
                    container.session.refreshProfile()
                }
                is ApiResult.Err -> _state.value =
                    _state.value.copy(error = result.failure.message)
            }
        }
    }
}

File: mobile/ios/Alma/Screens/Systems/PeopleScreen.swift
import SwiftUI

/// The people this account has saved. Compatibility needs at least one.
///
/// It also answers the finding that `SystemScreen(.compatibility)` silently
/// compared against `session.people.first` with no picker and no name shown —
/// fine with one saved person, wrong with three. The chosen person lives here,
/// in the session's own list order, and the compatibility screen names whoever
/// it read.
struct PeopleScreen: View {

    @Environment(AlmaSessionModel.self) private var session
    @Environment(AppRouter.self) private var router

    @State private var removing: Profile?

    var body: some View {
        ScreenScaffold(
            eyebrow: ScreenL10n.peopleEyebrow,
            title: ScreenL10n.peopleTitle,
            seed: 0x5045_4F50
        ) {
            Text(ScreenL10n.peopleLead).almaBody().almaReadingWidth()

            if session.people.isEmpty {
                NeedMore(
                    message: L10nCabinet.compatNeedsPerson,
                    actionTitle: L10nCabinet.addAPerson
                ) {
                    router.push(.addPerson)
                }
            } else {
                CabinetSection(label: ScreenL10n.peopleSaved, trailing: String(session.people.count)) {
                    ForEach(session.people) { person in
                        PersonRow(person: person) {
                            removing = person
                        }
                    }
                }

                Button { router.push(.addPerson) } label: {
                    Text(L10nCabinet.addAPerson)
                }
                .buttonStyle(.alma(.outline, fills: false))
                .padding(.top, AlmaMetrics.gapLarge)
            }

            // Whose birth data this is, and what we did about asking. The terms
            // ask the person entering it to have asked first; saying so at the
            // point of entry is worth more than saying it in a document.
            Text(ScreenL10n.peopleConsent)
                .almaMeta()
                .almaReadingWidth()
                .padding(.top, AlmaMetrics.gapLarge)
        }
        .task { await session.reloadProfiles() }
        // A birth is not something to delete on a single tap: every reading
        // keyed to it goes with it, and those are chapters somebody paid for.
        .confirmationDialog(
            Text(ScreenL10n.removePersonTitle),
            isPresented: Binding(get: { removing != nil }, set: { if !$0 { removing = nil } }),
            titleVisibility: .visible
        ) {
            Button(role: .destructive) {
                if let person = removing { Task { await remove(person) } }
            } label: {
                Text(ScreenL10n.removePerson)
            }
            Button(role: .cancel) { removing = nil } label: { Text(ScreenL10n.keep) }
        } message: {
            Text(ScreenL10n.removePersonWhat)
        }
    }

    private func remove(_ person: Profile) async {
        removing = nil
        try? await session.client.deleteProfile(id: person.id)
        await session.reloadProfiles()
        await session.refreshHub()
    }
}

/// One saved birth. Everything shown is what was entered — nothing computed,
/// nothing filled in.
private struct PersonRow: View {

    let person: Profile
    let remove: () -> Void

    var body: some View {
        AlmaRow {
            VStack(alignment: .leading, spacing: 4) {
                Text(verbatim: person.name?.isEmpty == false
                     ? person.name!
                     : String(localized: ScreenL10n.unnamedPerson))
                    .almaHeadingM()

                Text(verbatim: [
                    AlmaDate.readable(civil: person.birthDate),
                    person.birthTime.map { "\($0)" } ?? String(localized: L10nCabinet.unknownTime),
                    person.placeLabel,
                ].compactMap { $0 }.joined(separator: " · "))
                    .almaMeta()

                if let relation = person.relation, !relation.isEmpty {
                    Text(verbatim: relation).almaTag(.muted)
                }
            }
        } trailing: {
            Button(action: remove) {
                Text(ScreenL10n.remove)
                    .font(AlmaFonts.ui(13.5, weight: .medium))
                    .foregroundStyle(Color.almaDisagree.opacity(0.85))
            }
            .buttonStyle(.plain)
        }
    }
}

// MARK: — adding one

/// A second birth.
///
/// It asks the same five things the journey asks and in the same order, with two
/// differences that both come from what this is *for*: there is no ceremony
/// (nothing is being revealed, a comparison is being enabled), and it ends in
/// `session.addPerson(_:relation:)`, which sends `is_self: false` — the one call
/// in the app where that flag must be explicit, because "not said" on a second
/// birth would overwrite the account owner's own chart and every reading keyed
/// to it.
struct AddPersonScreen: View {

    @Environment(AlmaSessionModel.self) private var session
    @Environment(AppRouter.self) private var router
    @Environment(\.dismiss) private var dismiss

    @State private var name = ""
    @State private var relation = ""
    @State private var day: Int?
    @State private var month: Int?
    @State private var year: Int?
    @State private var hour: Int?
    @State private var minute: Int?
    @State private var meridiem: Meridiem?
    @State private var timeUnknown = false
    @State private var place: Place?
    @State private var query = ""
    @State private var results: [Place] = []
    @State private var saving = false
    @State private var failure: String?

    @FocusState private var typingPlace: Bool

    /// Whether this device writes hours 1–12 with AM/PM. The journey asks the
    /// same question; a Russian phone shows a 24-hour pair and no meridiem
    /// field at all — «АМ или РМ» on a Russian form was the owner's finding.
    private let twelveHour = DateFormatter
        .dateFormat(fromTemplate: "j", options: 0, locale: .current)?
        .contains("a") ?? true

    /// The same 92 years the journey offers, and the same argument against a
    /// `DatePicker`: a wheel always has a value, so it opens on today and a
    /// person who taps past it has silently told us this person was born this
    /// morning.
    private static let years: [Int] = {
        let thisYear = Calendar(identifier: .gregorian).component(.year, from: .now)
        return (0..<92).map { thisYear - 10 - $0 }
    }()

    var body: some View {
        ScreenScaffold(
            eyebrow: ScreenL10n.addPersonEyebrow,
            title: ScreenL10n.addPersonTitle,
            seed: 0x4144_4450
        ) {
            Text(ScreenL10n.addPersonLead).almaBody().almaReadingWidth()

            fields

            if let failure {
                Text(verbatim: failure)
                    .almaMeta()
                    .foregroundStyle(Color.almaDisagree)
                    .almaReadingWidth()
            }

            Button {
                Task { await save() }
            } label: {
                Text(saving ? ScreenL10n.saving : ScreenL10n.savePerson)
                    .lineLimit(1)
                    .minimumScaleFactor(0.75)
            }
            .buttonStyle(.almaGold)
            .disabled(!isComplete || saving)
            .padding(.top, AlmaMetrics.gapLarge)
            // The consent note stood here and the owner cut it.
        }
        .task(id: query) { await search() }
    }

    @ViewBuilder
    private var fields: some View {
        CabinetSection(label: ScreenL10n.theirName) {
            PlainField(text: $name, placeholder: ScreenL10n.namePlaceholder)
            PlainField(text: $relation, placeholder: ScreenL10n.relationPlaceholder)
        }

        CabinetSection(label: ScreenL10n.theirBirthday) {
            HStack(spacing: 9) {
                JourneyMenuField(
                    placeholder: JourneyL10n.dayShort, label: JourneyL10n.day,
                    options: Array(1...31), title: { String($0) }, selection: $day
                )
                JourneyMenuField(
                    placeholder: JourneyL10n.monthShort, label: JourneyL10n.month,
                    options: Array(1...12), title: { JourneyL10n.monthName($0) }, selection: $month
                )
                .frame(maxWidth: .infinity)
                JourneyMenuField(
                    placeholder: JourneyL10n.yearShort, label: JourneyL10n.year,
                    options: Self.years, title: { String($0) }, selection: $year
                )
            }

            if let day, let month, let year, !JourneyDate.isReal(day: day, month: month, year: year) {
                Text(JourneyL10n.impossibleDate)
                    .almaMeta()
                    .foregroundStyle(Color.almaGoldBright)
            }
        }

        CabinetSection(label: ScreenL10n.theirBirthTime) {
            HStack(spacing: 9) {
                JourneyMenuField(
                    placeholder: JourneyL10n.hourLabel, label: JourneyL10n.hourLabel,
                    options: twelveHour ? Array(1...12) : Array(0...23),
                    title: { String(format: "%02d", $0) }, selection: $hour
                )
                JourneyMenuField(
                    placeholder: JourneyL10n.minuteLabel, label: JourneyL10n.minuteLabel,
                    options: Array(0...59), title: { String(format: "%02d", $0) }, selection: $minute
                )
                if twelveHour {
                    JourneyMenuField(
                        placeholder: JourneyL10n.meridiemLabel, label: JourneyL10n.meridiemLabel,
                        options: [0, 1], title: { $0 == 0 ? "AM" : "PM" },
                        selection: Binding(
                            get: { meridiem.map { $0 == .am ? 0 : 1 } },
                            set: { meridiem = $0.map { $0 == 0 ? .am : .pm } }
                        )
                    )
                }
            }
            .opacity(timeUnknown ? 0.4 : 1)
            .disabled(timeUnknown)

            HStack(spacing: 14) {
                JourneyToggle(isOn: $timeUnknown, label: ScreenL10n.addPersonTimeUnknown)
                Text(ScreenL10n.addPersonTimeUnknown).almaMeta()
            }
            .padding(.top, 10)

            if timeUnknown {
                Text(JourneyL10n.lockedWithoutTime).almaMeta().almaReadingWidth()
            }
        }

        CabinetSection(label: ScreenL10n.theirBirthplace) {
            PlainField(text: $query, placeholder: JourneyL10n.placePlaceholder, focus: $typingPlace)

            ForEach(results.prefix(4)) { candidate in
                JourneySuggestionRow(place: candidate, selected: place?.id == candidate.id) {
                    place = candidate
                    query = candidate.label
                    results = []
                    typingPlace = false
                }
            }
        }
    }

    private var isComplete: Bool {
        day != nil && month != nil && year != nil && place != nil
            && (timeUnknown
                || (hour != nil && minute != nil && (!twelveHour || meridiem != nil)))
    }

    private func search() async {
        let trimmed = query.trimmingCharacters(in: .whitespacesAndNewlines)
        guard trimmed.count >= 2, trimmed != place?.label else {
            results = []
            return
        }
        // The same 180ms as the journey's step, and the same reason: `.task(id:)`
        // cancels the sleep and the request under it when the next keystroke
        // arrives, so a slow answer for "Mil" cannot land after a fast one for
        // "Milan".
        guard (try? await Task.sleep(for: .milliseconds(180))) != nil else { return }
        results = (try? await session.client.searchPlaces(trimmed)) ?? []
    }

    private func save() async {
        guard let day, let month, let year, let place, !saving else { return }
        saving = true
        defer { saving = false }
        failure = nil

        // The same calendar check the journey does, for the same reason: a 31st
        // of February reaches the backend as a 422 that would surface here as a
        // bare "something went wrong".
        guard JourneyDate.isReal(day: day, month: month, year: year) else {
            failure = String(localized: JourneyL10n.impossibleDate)
            return
        }

        var time: String?
        if !timeUnknown, let hour, let minute {
            if twelveHour, let meridiem {
                time = String(format: "%02d:%02d", meridiem.hour24(from: hour), minute)
            } else if !twelveHour {
                time = String(format: "%02d:%02d", hour, minute)
            }
        }

        let birth = BirthInput(
            birthDate: String(format: "%04d-%02d-%02d", year, month, day),
            birthTime: time,
            latitude: place.latitude,
            longitude: place.longitude,
            timezone: place.timezone,
            placeLabel: place.label,
            placeId: place.id,
            name: name.trimmingCharacters(in: .whitespaces).isEmpty ? nil : name
        )

        do {
            _ = try await session.addPerson(
                birth,
                relation: relation.trimmingCharacters(in: .whitespaces).isEmpty ? nil : relation
            )
            // Straight back to where the comparison lives. `dismiss()` popped
            // one screen and landed on the people list — a stop nobody asked
            // for; the owner's report was «нажать назад и опять в
            // совместимость». Both people screens come off the stack at once.
            var path = router.paths[router.tab] ?? []
            while let last = path.last, last == .addPerson || last == .people {
                path.removeLast()
            }
            router.paths[router.tab] = path
        } catch {
            failure = error.serverMessage ?? String(localized: error.displayText)
        }
    }
}

/// A plain capsule field. The journey's is the same shape but carries the gold
/// focus glow that belongs to a ceremony; a form in the cabinet is quieter.
private struct PlainField: View {

    @Binding var text: String
    let placeholder: LocalizedStringResource
    var focus: FocusState<Bool>.Binding?

    @FocusState private var owned: Bool

    var body: some View {
        Group {
            let field = TextField(
                "", text: $text,
                prompt: Text(placeholder).foregroundStyle(Color.almaMuted3)
            )
            .font(.almaBodyFont)
            .foregroundStyle(Color.almaInkLight)
            .tint(Color.almaGoldBright)
            .autocorrectionDisabled()

            if let focus { field.focused(focus) } else { field.focused($owned) }
        }
        .padding(.horizontal, 20)
        .frame(minHeight: AlmaMetrics.fieldHeight)
        .background(Capsule().fill(Color.almaVeil))
        .accessibilityLabel(Text(placeholder))
    }
}

#Preview {
    NavigationStack { PeopleScreen() }
        .environment(AlmaSessionModel.preview())
        .environment(AppRouter())
}
